File: BinarySearch/CheckPerfectSquare/perfectSquare.py
class Solution:
    def checkPerfectSquare(self, num) -> bool:
        root = int(num**0.5)
        if root**2 == num:
            return True
        return False
    def judgeSquareSum(self, c: int) -> bool:
        a = 0
        b = c
        while a<=b:
            if self.checkPerfectSquare(a) and self.checkPerfectSquare(b): return True
            elif not self.checkPerfectSquare(b):
                root = int(b**0.5)
                b = root**2
                a = c-b
            else:
                root = int(a**0.5)+1
                a = root**2
                b = c-a
        return False

        

File: BinarySearch/CheckPerfectSquare/perfectSquare.cpp
#include <cmath>

class Solution {
public:
    bool checkPerfectSquare(int num){
        int root = sqrt(num);
        if(root*root == num)return true;
        return false;
    }
    int makePerfectSquare(int n){
        int root = sqrt(n);
        return root*root;
    }
    bool judgeSquareSum(int c) {
        int a = 0;
        int b = c;
        while(a<=b){
            if(checkPerfectSquare(a) && checkPerfectSquare(b))return true;
            else if(!checkPerfectSquare(b)){
                b = makePerfectSquare(b);
                a = c-b;
            }else {
                a = (int)(sqrt(a)+1)*(int)(sqrt(a)+1);
                b = c-a;
            }
        }
        return false;
    }
};